### PR TITLE
add deleteAll, delete and getById

### PR DESCRIPTION
**Pull request by darkhanzhursin**
needs to test

### CODE REVIEW

**Comment by darkhanzhursin** [comment]
[BE: create baseApi functions and use it](https://trello.com/c/QAJHiZmQ/40-be-create-baseapi-functions-and-use-it)

File: server/api/index.js [hunk 1]
```diff
@@ -15,24 +15,47 @@ class Entity {
     }
   }
 
-  async getById() {}
+  async getById(req, res, entityName) {
+    try {
+      if (!req && !req.params && !req.params.id) {
```
Comment (eyesofdaveed) [inline server/api/index.js]: Install 'lodash' lib (version 4) and from now on use util functions from lodash.
`const entityId  = _.get(req, 'params.id')`

Comment (eyesofdaveed) [reply server/api/index.js]: `if (!entityId) return errorHandler({statusCode: 400, message:  '${entityName} ID required.' })`

File: server/api/index.js [hunk 2]
```diff
@@ -15,24 +15,47 @@ class Entity {
...
+        return res.status(400).json({ message: `${entityName} ID required.` });
+      }
+
+      const requestedEntity = await this.entityModel
+        .findOne({ _id: req.params.id })
```
Comment (eyesofdaveed) [inline server/api/index.js]: `_.get(req, ['params', 'id'])`

File: server/api/index.js [hunk 3]
```diff
@@ -15,24 +15,47 @@ class Entity {
...
-      }).skip(pageSizeInt * (pageNumberInt - 1)).limit(pageSizeInt);
-      return res
-      .json({ data: results });
+      const results = await this.entityModel
+        .find()
+        .sort({
+          createdAt: -1,
+        })
+        .skip(pageSizeInt * (pageNumberInt - 1))
+        .limit(pageSizeInt);
```
Comment (eyesofdaveed) [inline server/api/index.js:52]: separate with white line between logic

File: server/api/index.js [hunk 4]
```diff
@@ -15,24 +15,47 @@ class Entity {
...
+
+      res.json(requestedEntity);
+    } catch (err) {
+      errorHandler(err, req, res);
+    }
+  }
 
   async getAll({ req, res, pageSize = '25', pageNumber = '1' }) {
     try {
       const pageSizeInt = parseInt(pageSize);
```
Comment (eyesofdaveed) [inline server/api/index.js]: both `pageSize` and `pageNumber` will be set from `req.body.filter`

Comment (eyesofdaveed) [reply server/api/index.js]: `const pageSize = parseInt( _.get(req, 'body.filter', '25'));`

File: server/api/index.js [hunk 5]
```diff
@@ -45,9 +68,39 @@ class Entity {
     }
   }
 
-  async delete() {}
+  async delete(req, res, entityName) {
+    try {
+      if (!req && !req.params && !req.params.id) {
+        return res.status(400).json({ message: `${entityName} ID required.` });
+      }
```
Comment (eyesofdaveed) [inline server/api/index.js:89]: same as above

File: server/api/index.js [hunk 6]
```diff
@@ -45,9 +68,39 @@ class Entity {
     }
   }
 
-  async delete() {}
+  async delete(req, res, entityName) {
```
Comment (eyesofdaveed) [inline server/api/index.js:84]: `async delete({req, res, entityName}) {`

File: server/routes/users.js [hunk 7]
```diff
@@ -65,6 +65,31 @@ router.route('/').get(checkRole(ROLES.ADMIN), async (req, res) => {
...
+    } catch (err) {
+      errorHandler(err, req, res);
+    }
+  })
+  .get(async (req, res) => {
+    try {
+      await user.getById(req, res, modelName);
+    } catch (err) {
+      errorHandler(err, req, res);
+    }
```
Comment (eyesofdaveed) [inline server/routes/users.js:90]: This shouldn't belong here

File: server/api/index.js (hunk 6 above)
Comment (darkhanzhursin) [reply server/api/index.js:84]: > `async delete({req, res, entityName}) {`

what did u mean here?

**Review by DonJuvn — APPROVED** [review]

**Pull request merged**